src/languageservice/parser/yamlParser07.ts

```diff
@@ -33,7 +33,9 @@ export function parse(text: string, parserOptions: ParserOptions = defaultOption
   };
   const composer = new Composer(options);
   const lineCounter = new LineCounter();
-  const parser = new Parser(lineCounter.addNewLine);
+  const lexerTokensArr = Array.from(new Lexer().lex(text));
```

Comment:
I still don't like this, as we parse/tokenize file content twice just to detect end of file.
I think it would be much better/efficient  if `parse` function receive `TextDocument` object instead of `string`,
by using `TextDocument` you can receive last line content and make all validation that you need.

Reply:
Using TextDocument now I am checking with the last line empty/not